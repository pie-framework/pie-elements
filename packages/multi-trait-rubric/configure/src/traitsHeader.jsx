import React from 'react';
import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core/styles';
import IconButton from '@material-ui/core/IconButton';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import MoreVertIcon from '@material-ui/icons/MoreVert';
import { color } from '@pie-lib/render-ui';

import {
  Block,
  BlockWidth,
  PrimaryBlock,
  Row,
  SecondaryBlock,
  ScorePoint,
  MaxPointsPicker,
  SimpleInput,
  ScaleSettings,
  HeaderHeight,
  HeaderHeightLarge
} from './common';
import { labelPlugins } from './utils';

const styles = {
  label: {
    textAlign: 'center',
    width: '140px',
    border: 'none',
    margin: '8px',
    padding: '10px 0',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-around'
  },
  greyHeader: {
    background: color.secondaryBackground(),
    borderRadius: '4px',
    position: 'relative'
  },
  primaryBlockGreyHeader: {
    paddingTop: '12px'
  }
};

export class TraitsHeaderTile extends React.Component {
  state = {
    anchorEl: null,
  };

  UNSAFE_componentWillReceiveProps(nextProps) {
    if (nextProps.currentPosition !== this.props.currentPosition) {
      this.scrollToPosition(nextProps.currentPosition);
    }
  }

  onScorePointLabelChange = ({ scorePointLabel, value }) => {
    const { scorePointsLabels, onScaleChange } = this.props;

    if (value < 0 || value >= scorePointsLabels.length) return;

    scorePointsLabels[value] = scorePointLabel;

    onScaleChange({ scorePointsLabels });
  };

  handleClick = (event) => this.setState({ anchorEl: event.currentTarget });

  handleClose = () => this.setState({ anchorEl: null });

  scrollToPosition = position => this.secondaryBlock.scrollTo({ left: position });

  openMenu = () => {
    this.props.showDeleteScaleModal();
    this.handleClose();
  };

  render() {
    const {
      scorePointsValues,
      scorePointsLabels,
      traitLabel,
      classes,
      currentPosition,
      showStandards,
      onTraitLabelChange,
      showDescription,
      showLevelTagInput,
      maxPoints,
      updateMaxPointsFieldValue,
      scaleIndex,
      showScorePointLabels,
      secondaryBlockWidth,
      setSecondaryBlockRef,
    } = this.props;
    const { anchorEl } = this.state;

    return (
      <Row className={classes.greyHeader} height={showLevelTagInput ? HeaderHeightLarge : HeaderHeight}>
        <PrimaryBlock className={classes.primaryBlockGreyHeader}>
          {showLevelTagInput && (
            <SimpleInput
              markup={traitLabel || 'Trait'}
              onChange={onTraitLabelChange}
              pluginProps={labelPlugins}
              label='Level Label'
            />
          )}

          <ScaleSettings>
            <div>
              Scale {scaleIndex + 1}
            </div>

            <MaxPointsPicker
              maxPoints={maxPoints}
              onChange={updateMaxPointsFieldValue}
            />

            <div>
              <IconButton
                aria-label="more"
                aria-controls="long-menu"
                aria-haspopup="true"
                onClick={this.handleClick}
              >
                <MoreVertIcon/>
              </IconButton>
              <Menu
                id="long-menu"
                anchorEl={anchorEl}
                keepMounted
                open={!!anchorEl}
                onClose={this.handleClose}
              >
                {['Remove Scale'].map((option) => (
                  <MenuItem
                    key={option}
                    onClick={this.openMenu}
                  >
                    {option}
                  </MenuItem>
                ))}
              </Menu>
            </div>
          </ScaleSettings>
        </PrimaryBlock>

        <SecondaryBlock
          setRef={ref => {
            if (ref) {
              this.secondaryBlock = ref;
              setSecondaryBlockRef(ref);
            }
          }}
          width={`${secondaryBlockWidth}px`}
        >

          {showStandards && (
            <Block>
              <div className={classes.label}>
                Standard(s)
              </div>
            </Block>
          )}

          {showDescription && (
            <Block>
              <div className={classes.label}>
                Description
              </div>
            </Block>
          )}

          {scorePointsValues.map((scorePointsValue, index) => {
            const adjustedBlockWidth = BlockWidth + 2 * 8; // 8 is padding
            const remainingSpace = secondaryBlockWidth - adjustedBlockWidth * index + currentPosition - 128;
            const value = scorePointsValues.length - index - 1;
            let scoreDescriptor;

            try {
              scoreDescriptor = scorePointsLabels[value] || '';
            } catch (e) {
              scoreDescriptor = '';
            }

            return (
              <Block key={`secondary-block-part-${index}`}>
                <ScorePoint
                  scorePointsValue={scorePointsValue}
                  scoreDescriptor={scoreDescriptor}
                  pluginProps={labelPlugins}
                  showScorePointLabels={showScorePointLabels}
                  onChange={scorePointLabel => this.onScorePointLabelChange({ scorePointLabel, value })}
                  alignToRight={remainingSpace < 296} // 296 is the space required for the toolbar
                />
              </Block>
            )
          })}
        </SecondaryBlock>
      </Row>
    );
  }
}

TraitsHeaderTile.propTypes = {
  classes: PropTypes.object.isRequired,
  onTraitLabelChange: PropTypes.func,
  onScaleChange: PropTypes.func,
  scorePointsValues: PropTypes.arrayOf(PropTypes.number),
  scorePointsLabels: PropTypes.arrayOf(PropTypes.string),
  traitLabel: PropTypes.string,
  showStandards: PropTypes.bool,
  showLevelTagInput: PropTypes.bool,
  showDescription: PropTypes.bool,
  maxPoints: PropTypes.number,
  updateMaxPointsFieldValue: PropTypes.func,
  scaleIndex: PropTypes.number,
  currentPosition: PropTypes.number,
  secondaryBlockWidth: PropTypes.number,
  showDeleteScaleModal: PropTypes.func,
  showScorePointLabels: PropTypes.bool,
  setSecondaryBlockRef: PropTypes.func
};

export default withStyles(styles)(TraitsHeaderTile);
