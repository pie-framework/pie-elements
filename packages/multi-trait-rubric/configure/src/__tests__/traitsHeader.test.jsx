import { shallow } from 'enzyme';
import React from 'react';
import { TraitsHeaderTile } from '../traitsHeader';

describe('Trait Header', () => {
  let w;

  const wrapper = extras => {
    const defaults = {
      classes: {},
      onTraitLabelChange: () => {
      },
      onScaleChange: () => {
      },
      scorePointsValues: [0, 1, 2],
      scorePointsLabels: ['A', 'B', 'C'],
      traitLabel: 'Category',
      maxPoints: 2,
      scaleIndex: 0,
      currentPosition: 0,
      showStandards: true,
      showLevelTagInput: true,
      showDescription: true,
      showDeleteScaleModal: true,
      showScorePointLabels: true,
      ...extras
    };
    return shallow(<TraitsHeaderTile {...defaults} />);
  };

  describe('snapshot', () => {
    it('renders', () => {
      w = wrapper();

      expect(w).toMatchSnapshot();
    });

    it('renders without standards', () => {
      w = wrapper({ showStandards: false });

      expect(w).toMatchSnapshot();
    });

    it('renders without description', () => {
      w = wrapper({ showDescription: false });

      expect(w).toMatchSnapshot();
    });

    it('renders without score point labels', () => {
      w = wrapper({ showScorePointLabels: false });

      expect(w).toMatchSnapshot();
    });

    it('renders without level tag input', () => {
      w = wrapper({ showLevelTagInput: false });

      expect(w).toMatchSnapshot();
    });

    it('renders without score point values', () => {
      w = wrapper({ scorePointsValues: [] });

      expect(w).toMatchSnapshot();
    });
  });

  describe('logic', () => {
    let onScaleChange;
    const scrollToPositionSpy = jest.fn();

    beforeEach(() => {
      onScaleChange = jest.fn();
      w = wrapper({ onScaleChange });
    });

    describe('scroll position', () => {
      it ('does not change scroll position when current position prop does not change', () => {
        const wrap = wrapper({ currentPosition: 200 });

        expect(wrap.instance().props.currentPosition).toEqual(200);

        wrap.instance().scrollToPosition = scrollToPositionSpy;
        wrap.setProps({ currentPosition: 200 });

        expect(scrollToPositionSpy).not.toBeCalled();
      });

      it ('changes scroll position when current position prop changes', () => {
        const wrap = wrapper({ currentPosition: 200 });

        expect(wrap.instance().props.currentPosition).toEqual(200);

        wrap.instance().scrollToPosition = scrollToPositionSpy;
        wrap.setProps({ currentPosition: 300 });

        expect(scrollToPositionSpy).toBeCalledWith(300);
      });
    });

    describe('onScorePointLabelChange', () => {
      it('does not call onScaleChange with scorePointsLabels if value less than 0', () => {
        w.instance().onScorePointLabelChange({ scorePointLabel: 'New Label', value: -10 });

        expect(onScaleChange).not.toBeCalled();
      });

      it('does not call onScaleChange with scorePointsLabels if value more than length', () => {
        w.instance().onScorePointLabelChange({ scorePointLabel: 'New Label', value: 10 });

        expect(onScaleChange).not.toBeCalled();
      });

      it('call onScaleChange with scorePointsLabels', () => {
        const { scorePointsLabels } = w.instance().props;

        w.instance().onScorePointLabelChange({ scorePointLabel: 'New Label', value: 0 });

        expect(onScaleChange).toBeCalledWith({
          scorePointsLabels: [
            'New Label',
            ...scorePointsLabels.slice(1)
          ]
        });
      });
    });
  });
});
