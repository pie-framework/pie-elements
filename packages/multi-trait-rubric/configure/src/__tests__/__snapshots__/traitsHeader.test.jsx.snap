// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trait Header snapshot renders 1`] = `
<WithStyles(Component)
  height="160px"
>
  <WithStyles(Component)>
    <WithStyles(Component)
      label="Level Label"
      markup="Category"
      onChange={[Function]}
      pluginProps={
        Object {
          "audio": Object {
            "disabled": true,
          },
          "image": Object {
            "disabled": true,
          },
          "math": Object {
            "disabled": true,
          },
          "video": Object {
            "disabled": true,
          },
        }
      }
    />
    <WithStyles(Component)>
      <div>
        Scale 
        1
      </div>
      <WithStyles(Component)
        maxPoints={2}
      />
      <div>
        <WithStyles(IconButton)
          aria-controls="long-menu"
          aria-haspopup="true"
          aria-label="more"
          onClick={[Function]}
        >
          <pure(MoreVertIcon) />
        </WithStyles(IconButton)>
        <WithStyles(Menu)
          anchorEl={null}
          id="long-menu"
          keepMounted={true}
          onClose={[Function]}
          open={false}
        >
          <WithStyles(MenuItem)
            key="Remove Scale"
            onClick={[Function]}
          >
            Remove Scale
          </WithStyles(MenuItem)>
        </WithStyles(Menu)>
      </div>
    </WithStyles(Component)>
  </WithStyles(Component)>
  <WithStyles(Component)
    setRef={[Function]}
  >
    <WithStyles(Component)>
      <div>
        Standard(s)
      </div>
    </WithStyles(Component)>
    <WithStyles(Component)>
      <div>
        Description
      </div>
    </WithStyles(Component)>
    <WithStyles(Component)
      key="secondary-block-part-0"
    >
      <WithStyles(Component)
        onChange={[Function]}
        pluginProps={
          Object {
            "audio": Object {
              "disabled": true,
            },
            "image": Object {
              "disabled": true,
            },
            "math": Object {
              "disabled": true,
            },
            "video": Object {
              "disabled": true,
            },
          }
        }
        scoreDescriptor="C"
        scorePointsValue={0}
        showScorePointLabels={true}
      />
    </WithStyles(Component)>
    <WithStyles(Component)
      key="secondary-block-part-1"
    >
      <WithStyles(Component)
        onChange={[Function]}
        pluginProps={
          Object {
            "audio": Object {
              "disabled": true,
            },
            "image": Object {
              "disabled": true,
            },
            "math": Object {
              "disabled": true,
            },
            "video": Object {
              "disabled": true,
            },
          }
        }
        scoreDescriptor="B"
        scorePointsValue={1}
        showScorePointLabels={true}
      />
    </WithStyles(Component)>
    <WithStyles(Component)
      key="secondary-block-part-2"
    >
      <WithStyles(Component)
        onChange={[Function]}
        pluginProps={
          Object {
            "audio": Object {
              "disabled": true,
            },
            "image": Object {
              "disabled": true,
            },
            "math": Object {
              "disabled": true,
            },
            "video": Object {
              "disabled": true,
            },
          }
        }
        scoreDescriptor="A"
        scorePointsValue={2}
        showScorePointLabels={true}
      />
    </WithStyles(Component)>
  </WithStyles(Component)>
</WithStyles(Component)>
`;

exports[`Trait Header snapshot renders without description 1`] = `
<WithStyles(Component)
  height="160px"
>
  <WithStyles(Component)>
    <WithStyles(Component)
      label="Level Label"
      markup="Category"
      onChange={[Function]}
      pluginProps={
        Object {
          "audio": Object {
            "disabled": true,
          },
          "image": Object {
            "disabled": true,
          },
          "math": Object {
            "disabled": true,
          },
          "video": Object {
            "disabled": true,
          },
        }
      }
    />
    <WithStyles(Component)>
      <div>
        Scale 
        1
      </div>
      <WithStyles(Component)
        maxPoints={2}
      />
      <div>
        <WithStyles(IconButton)
          aria-controls="long-menu"
          aria-haspopup="true"
          aria-label="more"
          onClick={[Function]}
        >
          <pure(MoreVertIcon) />
        </WithStyles(IconButton)>
        <WithStyles(Menu)
          anchorEl={null}
          id="long-menu"
          keepMounted={true}
          onClose={[Function]}
          open={false}
        >
          <WithStyles(MenuItem)
            key="Remove Scale"
            onClick={[Function]}
          >
            Remove Scale
          </WithStyles(MenuItem)>
        </WithStyles(Menu)>
      </div>
    </WithStyles(Component)>
  </WithStyles(Component)>
  <WithStyles(Component)
    setRef={[Function]}
  >
    <WithStyles(Component)>
      <div>
        Standard(s)
      </div>
    </WithStyles(Component)>
    <WithStyles(Component)
      key="secondary-block-part-0"
    >
      <WithStyles(Component)
        onChange={[Function]}
        pluginProps={
          Object {
            "audio": Object {
              "disabled": true,
            },
            "image": Object {
              "disabled": true,
            },
            "math": Object {
              "disabled": true,
            },
            "video": Object {
              "disabled": true,
            },
          }
        }
        scoreDescriptor="C"
        scorePointsValue={0}
        showScorePointLabels={true}
      />
    </WithStyles(Component)>
    <WithStyles(Component)
      key="secondary-block-part-1"
    >
      <WithStyles(Component)
        onChange={[Function]}
        pluginProps={
          Object {
            "audio": Object {
              "disabled": true,
            },
            "image": Object {
              "disabled": true,
            },
            "math": Object {
              "disabled": true,
            },
            "video": Object {
              "disabled": true,
            },
          }
        }
        scoreDescriptor="B"
        scorePointsValue={1}
        showScorePointLabels={true}
      />
    </WithStyles(Component)>
    <WithStyles(Component)
      key="secondary-block-part-2"
    >
      <WithStyles(Component)
        onChange={[Function]}
        pluginProps={
          Object {
            "audio": Object {
              "disabled": true,
            },
            "image": Object {
              "disabled": true,
            },
            "math": Object {
              "disabled": true,
            },
            "video": Object {
              "disabled": true,
            },
          }
        }
        scoreDescriptor="A"
        scorePointsValue={2}
        showScorePointLabels={true}
      />
    </WithStyles(Component)>
  </WithStyles(Component)>
</WithStyles(Component)>
`;

exports[`Trait Header snapshot renders without level tag input 1`] = `
<WithStyles(Component)
  height="100px"
>
  <WithStyles(Component)>
    <WithStyles(Component)>
      <div>
        Scale 
        1
      </div>
      <WithStyles(Component)
        maxPoints={2}
      />
      <div>
        <WithStyles(IconButton)
          aria-controls="long-menu"
          aria-haspopup="true"
          aria-label="more"
          onClick={[Function]}
        >
          <pure(MoreVertIcon) />
        </WithStyles(IconButton)>
        <WithStyles(Menu)
          anchorEl={null}
          id="long-menu"
          keepMounted={true}
          onClose={[Function]}
          open={false}
        >
          <WithStyles(MenuItem)
            key="Remove Scale"
            onClick={[Function]}
          >
            Remove Scale
          </WithStyles(MenuItem)>
        </WithStyles(Menu)>
      </div>
    </WithStyles(Component)>
  </WithStyles(Component)>
  <WithStyles(Component)
    setRef={[Function]}
  >
    <WithStyles(Component)>
      <div>
        Standard(s)
      </div>
    </WithStyles(Component)>
    <WithStyles(Component)>
      <div>
        Description
      </div>
    </WithStyles(Component)>
    <WithStyles(Component)
      key="secondary-block-part-0"
    >
      <WithStyles(Component)
        onChange={[Function]}
        pluginProps={
          Object {
            "audio": Object {
              "disabled": true,
            },
            "image": Object {
              "disabled": true,
            },
            "math": Object {
              "disabled": true,
            },
            "video": Object {
              "disabled": true,
            },
          }
        }
        scoreDescriptor="C"
        scorePointsValue={0}
        showScorePointLabels={true}
      />
    </WithStyles(Component)>
    <WithStyles(Component)
      key="secondary-block-part-1"
    >
      <WithStyles(Component)
        onChange={[Function]}
        pluginProps={
          Object {
            "audio": Object {
              "disabled": true,
            },
            "image": Object {
              "disabled": true,
            },
            "math": Object {
              "disabled": true,
            },
            "video": Object {
              "disabled": true,
            },
          }
        }
        scoreDescriptor="B"
        scorePointsValue={1}
        showScorePointLabels={true}
      />
    </WithStyles(Component)>
    <WithStyles(Component)
      key="secondary-block-part-2"
    >
      <WithStyles(Component)
        onChange={[Function]}
        pluginProps={
          Object {
            "audio": Object {
              "disabled": true,
            },
            "image": Object {
              "disabled": true,
            },
            "math": Object {
              "disabled": true,
            },
            "video": Object {
              "disabled": true,
            },
          }
        }
        scoreDescriptor="A"
        scorePointsValue={2}
        showScorePointLabels={true}
      />
    </WithStyles(Component)>
  </WithStyles(Component)>
</WithStyles(Component)>
`;

exports[`Trait Header snapshot renders without score point labels 1`] = `
<WithStyles(Component)
  height="160px"
>
  <WithStyles(Component)>
    <WithStyles(Component)
      label="Level Label"
      markup="Category"
      onChange={[Function]}
      pluginProps={
        Object {
          "audio": Object {
            "disabled": true,
          },
          "image": Object {
            "disabled": true,
          },
          "math": Object {
            "disabled": true,
          },
          "video": Object {
            "disabled": true,
          },
        }
      }
    />
    <WithStyles(Component)>
      <div>
        Scale 
        1
      </div>
      <WithStyles(Component)
        maxPoints={2}
      />
      <div>
        <WithStyles(IconButton)
          aria-controls="long-menu"
          aria-haspopup="true"
          aria-label="more"
          onClick={[Function]}
        >
          <pure(MoreVertIcon) />
        </WithStyles(IconButton)>
        <WithStyles(Menu)
          anchorEl={null}
          id="long-menu"
          keepMounted={true}
          onClose={[Function]}
          open={false}
        >
          <WithStyles(MenuItem)
            key="Remove Scale"
            onClick={[Function]}
          >
            Remove Scale
          </WithStyles(MenuItem)>
        </WithStyles(Menu)>
      </div>
    </WithStyles(Component)>
  </WithStyles(Component)>
  <WithStyles(Component)
    setRef={[Function]}
  >
    <WithStyles(Component)>
      <div>
        Standard(s)
      </div>
    </WithStyles(Component)>
    <WithStyles(Component)>
      <div>
        Description
      </div>
    </WithStyles(Component)>
    <WithStyles(Component)
      key="secondary-block-part-0"
    >
      <WithStyles(Component)
        onChange={[Function]}
        pluginProps={
          Object {
            "audio": Object {
              "disabled": true,
            },
            "image": Object {
              "disabled": true,
            },
            "math": Object {
              "disabled": true,
            },
            "video": Object {
              "disabled": true,
            },
          }
        }
        scoreDescriptor="C"
        scorePointsValue={0}
        showScorePointLabels={false}
      />
    </WithStyles(Component)>
    <WithStyles(Component)
      key="secondary-block-part-1"
    >
      <WithStyles(Component)
        onChange={[Function]}
        pluginProps={
          Object {
            "audio": Object {
              "disabled": true,
            },
            "image": Object {
              "disabled": true,
            },
            "math": Object {
              "disabled": true,
            },
            "video": Object {
              "disabled": true,
            },
          }
        }
        scoreDescriptor="B"
        scorePointsValue={1}
        showScorePointLabels={false}
      />
    </WithStyles(Component)>
    <WithStyles(Component)
      key="secondary-block-part-2"
    >
      <WithStyles(Component)
        onChange={[Function]}
        pluginProps={
          Object {
            "audio": Object {
              "disabled": true,
            },
            "image": Object {
              "disabled": true,
            },
            "math": Object {
              "disabled": true,
            },
            "video": Object {
              "disabled": true,
            },
          }
        }
        scoreDescriptor="A"
        scorePointsValue={2}
        showScorePointLabels={false}
      />
    </WithStyles(Component)>
  </WithStyles(Component)>
</WithStyles(Component)>
`;

exports[`Trait Header snapshot renders without score point values 1`] = `
<WithStyles(Component)
  height="160px"
>
  <WithStyles(Component)>
    <WithStyles(Component)
      label="Level Label"
      markup="Category"
      onChange={[Function]}
      pluginProps={
        Object {
          "audio": Object {
            "disabled": true,
          },
          "image": Object {
            "disabled": true,
          },
          "math": Object {
            "disabled": true,
          },
          "video": Object {
            "disabled": true,
          },
        }
      }
    />
    <WithStyles(Component)>
      <div>
        Scale 
        1
      </div>
      <WithStyles(Component)
        maxPoints={2}
      />
      <div>
        <WithStyles(IconButton)
          aria-controls="long-menu"
          aria-haspopup="true"
          aria-label="more"
          onClick={[Function]}
        >
          <pure(MoreVertIcon) />
        </WithStyles(IconButton)>
        <WithStyles(Menu)
          anchorEl={null}
          id="long-menu"
          keepMounted={true}
          onClose={[Function]}
          open={false}
        >
          <WithStyles(MenuItem)
            key="Remove Scale"
            onClick={[Function]}
          >
            Remove Scale
          </WithStyles(MenuItem)>
        </WithStyles(Menu)>
      </div>
    </WithStyles(Component)>
  </WithStyles(Component)>
  <WithStyles(Component)
    setRef={[Function]}
  >
    <WithStyles(Component)>
      <div>
        Standard(s)
      </div>
    </WithStyles(Component)>
    <WithStyles(Component)>
      <div>
        Description
      </div>
    </WithStyles(Component)>
  </WithStyles(Component)>
</WithStyles(Component)>
`;

exports[`Trait Header snapshot renders without standards 1`] = `
<WithStyles(Component)
  height="160px"
>
  <WithStyles(Component)>
    <WithStyles(Component)
      label="Level Label"
      markup="Category"
      onChange={[Function]}
      pluginProps={
        Object {
          "audio": Object {
            "disabled": true,
          },
          "image": Object {
            "disabled": true,
          },
          "math": Object {
            "disabled": true,
          },
          "video": Object {
            "disabled": true,
          },
        }
      }
    />
    <WithStyles(Component)>
      <div>
        Scale 
        1
      </div>
      <WithStyles(Component)
        maxPoints={2}
      />
      <div>
        <WithStyles(IconButton)
          aria-controls="long-menu"
          aria-haspopup="true"
          aria-label="more"
          onClick={[Function]}
        >
          <pure(MoreVertIcon) />
        </WithStyles(IconButton)>
        <WithStyles(Menu)
          anchorEl={null}
          id="long-menu"
          keepMounted={true}
          onClose={[Function]}
          open={false}
        >
          <WithStyles(MenuItem)
            key="Remove Scale"
            onClick={[Function]}
          >
            Remove Scale
          </WithStyles(MenuItem)>
        </WithStyles(Menu)>
      </div>
    </WithStyles(Component)>
  </WithStyles(Component)>
  <WithStyles(Component)
    setRef={[Function]}
  >
    <WithStyles(Component)>
      <div>
        Description
      </div>
    </WithStyles(Component)>
    <WithStyles(Component)
      key="secondary-block-part-0"
    >
      <WithStyles(Component)
        onChange={[Function]}
        pluginProps={
          Object {
            "audio": Object {
              "disabled": true,
            },
            "image": Object {
              "disabled": true,
            },
            "math": Object {
              "disabled": true,
            },
            "video": Object {
              "disabled": true,
            },
          }
        }
        scoreDescriptor="C"
        scorePointsValue={0}
        showScorePointLabels={true}
      />
    </WithStyles(Component)>
    <WithStyles(Component)
      key="secondary-block-part-1"
    >
      <WithStyles(Component)
        onChange={[Function]}
        pluginProps={
          Object {
            "audio": Object {
              "disabled": true,
            },
            "image": Object {
              "disabled": true,
            },
            "math": Object {
              "disabled": true,
            },
            "video": Object {
              "disabled": true,
            },
          }
        }
        scoreDescriptor="B"
        scorePointsValue={1}
        showScorePointLabels={true}
      />
    </WithStyles(Component)>
    <WithStyles(Component)
      key="secondary-block-part-2"
    >
      <WithStyles(Component)
        onChange={[Function]}
        pluginProps={
          Object {
            "audio": Object {
              "disabled": true,
            },
            "image": Object {
              "disabled": true,
            },
            "math": Object {
              "disabled": true,
            },
            "video": Object {
              "disabled": true,
            },
          }
        }
        scoreDescriptor="A"
        scorePointsValue={2}
        showScorePointLabels={true}
      />
    </WithStyles(Component)>
  </WithStyles(Component)>
</WithStyles(Component)>
`;
